staging/src/k8s.io/apiserver/pkg/storage/cacher/cacher.go

```diff
@@ -975,6 +975,10 @@ func (c *Cacher) GuaranteedUpdate(
 
 // Count implements storage.Interface.
 func (c *Cacher) Count(pathPrefix string) (int64, error) {
+	cnt := len(c.watchCache.ListKeys())
```

Comment:
what are the implications to apf and metrics if this count becomes persistently stale?

Reply:
you means watchCache is inconsistent, or there is a continuous small version gap between the watchCache observed and the latest version in etcd?

Here is some of my throught, correct me if I'm wrong.
1. For APF: 
  I think that the inconsistent case doesn't in the scope of this PR to disscuss. Because the APF doestn't in the critical path. Once it happens, the APF still works, just less accurate and less fair. For the case of a small version gap between cache and etcd, as pervious said, this count doesn't need to be very accurate, so I think it's ok..

2. For metrics:
   To the inconsistent scenario, this could make people confused to why it's different with etcd. But I doubt that is anyone using this metric to report watchCache is inconsistent? So IMO, the implication is minor. Also, for version gap case, monitoring may appear to have a slight delay because they are eventually consistent.

Reply:
I think that's *probably* ok, since this is not exposed for API users to directly call (they can only observe via metrics), but the docs need to be *super* clear that this is not guaranteed to be a live lookup

Reply:
The Count function was only used by metrics, and after add ready check, it has returned more exactly.